src/api/student.js

```diff
@@ -82,22 +81,16 @@ app.post('/login', (req, res) => {
         res.failureJson('A student could not be found with this email address.');
         return;
     }
-    if (!student.confirmedStudentEmail) {
-        res.failureJson('Please check your student email inbox to verify your address before logging in.');
-        return;
-    }
-    if (!student.confirmedBackupEmail) {
-        res.failureJson('Please check your backup email inbox to verify your address before logging in.');
-        return;
-    }
-    let salt = student.salt;
-    let hashedPassword = saltAndHash(password, salt);
-    if (hashedPassword !== student.hashedPassword) {
-        res.failureJson('Incorrect password. Please try again.');
-    }
     let authTokenA = createToken(); // token for valid credentials
     let authTokenB = createToken(); // tokens sent to email address
-    return sendAuthenticationToken(student.studentEmail, authTokenB).then(() => {
+    verifyPassword(password, student.hashedPassword).then(match => {
+        if (!match) {
+            res.failureJson('Incorrect password. Please try again.');
```

Comment:
I would rather throw an error here and leave the `catch` at the end of the promise chain to handle this.